### PR TITLE
AIRO-0000 Remove the UNITY_2020_1_OR_NEWER & UNITY_2020_1 defines 

### PR DESCRIPTION
**Pull request by K-Tone**
... and paths leading to them. We have ArticulationBody support in any Unity release since 20.3 LTS, no one in robotics should be using the Rigidbody path instead.

This is a preliminary PR, intended to land after discussion with Vidur. 

## Proposed change(s)

Describe the changes made in this PR.

### Useful links (GitHub issues, JIRA tickets, forum threads, etc.)

Provide any relevant links here.

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [x] Code refactor
- [ ] Documentation update
- [ ] Other (please describe)

## Testing and Verification

Please describe the tests that you ran to verify your changes. Please also provide instructions, ROS packages, and Unity project files as appropriate so we can reproduce the test environment. 

### Test Configuration:
- Unity Version: [e.g. Unity 2020.2.0f1]
- Unity machine OS + version: [e.g. Windows 10]
- ROS machine OS + version: [e.g. Ubuntu 18.04, ROS Noetic]
- ROS–Unity communication: [e.g. Docker]

## Checklist
- [x] Ensured this PR is up-to-date with the `dev` branch
- [x] Created this PR to target the `dev` branch
- [ ] Followed the style guidelines as described in the [Contribution Guidelines](https://github.com/Unity-Technologies/URDF-Importer/blob/main/CONTRIBUTING.md)
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Increased the [test coverage criteria](https://github.com/Unity-Technologies/URDF-Importer/blob/dev/.yamato/yamato-config.yml#L18) by 3%
- [ ] Updated the [Changelog](https://github.com/Unity-Technologies/URDF-Importer/blob/dev/com.unity.robotics.urdf-importer/CHANGELOG.md) and described changes in the [Unreleased section](https://github.com/Unity-Technologies/URDF-Importer/blob/dev/com.unity.robotics.urdf-importer/CHANGELOG.md#unreleased)
- [ ] Updated the documentation as appropriate

## Other comments

### CODE REVIEW

**Comment by unity-cla-assistant** [comment]
[CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/URDF-Importer?pullRequest=191) <br/>All committers have signed the CLA.

File: TestUrdfImporter/ProjectSettings/ProjectVersion.txt
Comment (K-Tone) [inline TestUrdfImporter/ProjectSettings/ProjectVersion.txt:2]: Updated to a newer Unity LTS

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfInertial.cs
Comment (K-Tone) [inline com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfInertial.cs]: @vidurvij-Unity ^^

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJoint.cs
Comment (K-Tone) [inline com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJoint.cs]: @vidurvij-Unity ^^

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointFloating.cs
Comment (K-Tone) [inline com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointFloating.cs:35]: @vidurvij-Unity ^^

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointPlanar.cs
Comment (K-Tone) [inline com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointPlanar.cs]: @vidurvij-Unity ^^

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointRevolute.cs
Comment (K-Tone) [inline com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointRevolute.cs:58]: I think this won't work, this property is only for setting force at the moment, won't read back.

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfInertial.cs
Comment (vidurvij-Unity) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfInertial.cs]: I think we added this a long time ago in one of our debugging sessions. This is no longer needed.

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJoint.cs
Comment (vidurvij-Unity) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJoint.cs]: This is a remnant of ROS#. They had that API for rigid body but I went through the code and I think the axisofMotion is always defined.

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointFloating.cs
Comment (vidurvij-Unity) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointFloating.cs:35]: This was an oversight. But we need to make changes to correct this. Since this is a planar joint we need to return a pari of floats.

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointPlanar.cs
Comment (vidurvij-Unity) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointPlanar.cs]: Same reasoning as above. This should be removed.

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointPrismatic.cs
Comment (vidurvij-Unity) [inline com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointPrismatic.cs:24]: Was this not being used anywhere ?

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointRevolute.cs
Comment (vidurvij-Unity) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointRevolute.cs:58]: Is there a corresponding API to read force ?

**Review by vidurvij-Unity — COMMENTED** [review]
Left comments for you to look at.

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfInertial.cs
Comment (K-Tone) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfInertial.cs]: Ok got you, I'll cut this code then.

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJoint.cs
Comment (K-Tone) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJoint.cs]: Right, so we should probably simplify it then since it's always going to be returning true

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointFloating.cs
Comment (K-Tone) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointFloating.cs:35]: Maybe we could collab on a proper way to fix this? I don't directly see what values to return as a pair...

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointPlanar.cs
Comment (K-Tone) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointPlanar.cs]: Got it. Will do.

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointPrismatic.cs
Comment (K-Tone) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointPrismatic.cs:24]: I think that we didn't have the max linear velocity property in the initial release of ABs back in 2020.1. It appears later, in 2020.2. I'm cutting this compatibility layer because 2020.1 is an obsolete tech stream pre-empted by 2020.3 LTS. No customer should be using it, and we don't need to support it any more.

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointRevolute.cs
Comment (K-Tone) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointRevolute.cs:58]: I think this is related to the concept of inverse dynamics, which is only available starting 2022.1. @JuozasK

Comment (JuozasK) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointRevolute.cs:58]: Yeah, for now jointForce will only return what you set there before yourself. As for getting actual drive forces or related counteracting forces, that API landed in 2022.1.0a16

There's work on improving the Inverse Dynamics API with accumulated forces and an external forces getter which would allow to get the force in reduced space needed to counteract any previously applied forces in general space

Comment (K-Tone) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointRevolute.cs:58]: I think if we were to solve this, someone should open a separate ticket. I'll leave this to Vidur. Clearly out of scope for the purposes of this PR.

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointFloating.cs
Comment (K-Tone) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointFloating.cs:35]: I think we should have a discussion over zoom or slack on how to move this forward really. 

Clearly, GetPosition() / GetVelocity() / GetEffort() interface is limiting and, in a way, confusing. Once we work with ArtculationBodies, any getter should return an instance of ArticulationReducedSpace actually instead of a float. This will be float for revolute, float for prismatic and up to three floats for spherical. I don't yet understand how we map urdf to articulation, but my intention would be to expose ABs directly here and not try to insert an abstracting interface, as that one is only an obstacle from my pov. 

Having that said, I would suggest to open a ticket to redesign this bit. The behaviour is not broken nor improved by this PR.

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfInertial.cs
Comment (JuozasK) [inline com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfInertial.cs:18]: For future reference, we will be wanting to update the namespaces as per Robotics -> Simulation namespace change

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfInertial.cs
Comment (JuozasK) [inline com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfInertial.cs:102]: Is it just me, or does the comment not make sense? ABs have a public property called inertiaTensor and inertiaTensorRotation?

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfInertial.cs
Comment (JuozasK) [inline com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfInertial.cs:209]: nitpick, but I think it's "lambda" not "lamda"?

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointContinuous.cs
Comment (JuozasK) [inline com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointContinuous.cs:55]: Is there a point to this method, as ArticulationBody.jointForce will not return anything unless explicitly set by the user?

**Review by JuozasK — DISMISSED** [review]
Generally everything looks good, left some minor comments that may or may not be in scope for this PR

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfInertial.cs
Comment (K-Tone) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfInertial.cs:18]: Thanks. But out of the scope of this PR.

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfInertial.cs
Comment (K-Tone) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfInertial.cs:102]: We didn't have it back in 2020.1 (in the initial AB roll out).

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfInertial.cs
Comment (K-Tone) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfInertial.cs:209]: This is actually a good catch, I'll fix this up in a batch if Vidur requests more changes :)

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointContinuous.cs
Comment (K-Tone) [reply com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointContinuous.cs:55]: Yeah really. I think this whole GetXXX thing needs to be redesigned, but that should be another card and another scope of work. There's quite a few that I'd change in the URDF importer.

**Comment by JuozasK** [comment]
One thing I just noticed, URDF Robot script still says "All rigidbodies" in the editor. We should rename that to "All Articulation bodies" 
image

**Review by JuozasK — APPROVED** [review]
Looks good!

**Comment by vidurvij-Unity** [comment]
Please add the relevant jira ticket in the description of the pull request.

**Review by vidurvij-Unity — APPROVED** [review]

**Comment by K-Tone** [comment]
Jira link: https://jira.unity3d.com/browse/PHYS-135

**Pull request merged**
